src/java.base/share/classes/java/math/BigDecimal.java

```diff
@@ -4261,23 +4210,30 @@ private String layoutChars(boolean sci) {
         if (scale == 2  &&
             intCompact >= 0 && intCompact < Integer.MAX_VALUE) {
             // currency fast path
-            int lowInt = (int)intCompact % 100;
             int highInt = (int)intCompact / 100;
-            return (Integer.toString(highInt) + '.' +
-                    StringBuilderHelper.DIGIT_TENS[lowInt] +
-                    StringBuilderHelper.DIGIT_ONES[lowInt]) ;
+            int lowInt = (int) intCompact - highInt * 100;
+            int highIntSize = JLA.stringSize(highInt);
+            byte[] buf = new byte[highIntSize + 3];
```

Comment:
I think the String concat indy call site already generates code that computes the size of the char array to be allocated (remember the lengthCoder from String Concat). We can probably just use `highInt + '.' + StringBuilderHelper.DIGIT_TENS[lowInt] + StringBuilderHelper.DIGIT_ONES[lowInt]`.

Reply:
> I think the String concat indy call site already generates code that computes the size of the char array to be allocated (remember the lengthCoder from String Concat). We can probably just use `highInt + '.' + StringBuilderHelper.DIGIT_TENS[lowInt] + StringBuilderHelper.DIGIT_ONES[lowInt]`.

The code of StringBuilderHelper is duplicated, I want to remove it